openslides_backend/models/check_models.py

```diff
@@ -0,0 +1,690 @@
+            if self.is_normal_field(x) or self.is_template_field(x)
+        )
+        collection_fields_types = (
+            x
+            for x in self.models[collection]().get_fields()
+            if self.is_normal_field(x.own_field_name)
+            or self.is_template_field(x.own_field_name)
+        )
+        collection_fields = set(
+            field.own_field_name
+            if not isinstance(field, BaseTemplateField)
+            else field.get_structured_field_name("")
+            for field in collection_fields_types
+        )
+
+        errors = False
+        if collection_fields - model_fields:
+            error = f"{collection}/{model['id']}: Missing fields {', '.join(collection_fields - model_fields)}"
+            self.errors.append(error)
+            errors = True
+        if model_fields - collection_fields:
+            error = f"{collection}/{model['id']}: Invalid fields {', '.join(model_fields - collection_fields)}"
+            self.errors.append(error)
+            errors = True
+        return errors
+
+    def check_template_fields(self, model: Dict[str, Any], collection: str) -> bool:
+        """
+        Only checks that for each replacement a structured field exists and
+        not too many structured fields. Does not check the content.
+        Returns True on errors.
+        """
+        errors = False
+        for template_field in self.get_fields(collection):
+            if not isinstance(template_field, BaseTemplateField):
+                continue
+            field_error = False
+            replacements = model.get(template_field.get_structured_field_name(""))
+
+            if not isinstance(replacements, list):
+                self.errors.append(
+                    f"{collection}/{model['id']}/{template_field.own_field_name}: Replacements for the template field must be a list"
+                )
+                field_error = True
+                continue
+            for replacement in replacements:
+                if not isinstance(replacement, str):
+                    self.errors.append(
+                        f"{collection}/{model['id']}/{template_field.own_field_name}: Each replacement for the template field must be a string"
+                    )
+                    field_error = True
+            if field_error:
+                errors = True
+                continue
+            replacement_collection = None
+            if template_field.replacement_collection:
+                replacement_collection = (
+                    template_field.replacement_collection.collection
+                )
+
+            for replacement in replacements:
+                structured_field = self.make_structured(template_field, replacement)
+                if structured_field not in model:
+                    self.errors.append(
+                        f"{collection}/{model['id']}/{template_field.own_field_name}: Missing {structured_field} since it is given as a replacement"
```

Comment:
`own_field_name` contains the name without the `$`, better use `get_template_field_name()`

Reply:
Use get_own_field_name here.